### PR TITLE
Use token request to sync sa

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: pkg/helpers/sa_syncer.go [hunk 1]
```diff
@@ -16,108 +17,140 @@ import (
...
+			}
+
+			return saToken, nil
 		}
 
-		saToken, ok := tokenSecret.Data["token"]
-		if !ok {
-			return fmt.Errorf("no token in data for secret %s", tokenSecretName)
+		// 8640 hour
+		expirationSec := int64(8640 * 3600)
```
Comment (zhujian7) [inline pkg/helpers/sa_syncer.go]: what will happen after this period? will the klusterlet refresh the token?

**Comment by xuezhaojun** [comment]
LGTM

File: pkg/helpers/sa_syncer.go (hunk 1 above)
Comment (qiujian16) [reply pkg/helpers/sa_syncer.go]: I add a TODO to check the validity of the token and refresh if necessary

**Comment by qiujian16** [comment]
/hold

still need to add rbac setting.

File: pkg/helpers/sa_syncer.go [hunk 2]
```diff
@@ -2,122 +2,183 @@ package helpers
...
+		if lifetime < refreshThreshold {
+			return false
+		}
 	}
 
-	return fmt.Errorf("no token secret for this service account %s", sa.Name)
+	return true
 }
 
 // RenderToKubeconfigSecret would render saToken to a secret.
```
Comment (zhujian7) [inline pkg/helpers/sa_syncer.go]: // applyKubeconfigSecret ....

**Pull request merged**